### ISSUE TITLE
Polygon mumbai bor.milestone=true invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by mikolajsobolewski** [ledgerwatch/erigon#8338]
#### System information

Erigon version: `.erigon version 2.50.1-c91efdcb

OS & Version: Ubuntu 22.04.3 LTS

Commit hash: c91efdcb2e885089d95a551c5ec7fa440d53d156

Erigon Command (with flags/config):
````
--chain=mumbai
--bor.heimdall=http://heimdallr-mumbai:1317
--bor.milestone=true
--datadir=/data
--http=true
--http.addr=0.0.0.0
--http.compression=true
--http.vhosts=*
--http.corsdomain=*
--ws=true
--ws.compression=true
--prune=hrtc
--prune.h.older=15768000
--prune.r.older=15768000
--prune.t.older=15768000
--prune.c.older=15768000
--snapshots=false
--bootnodes=enode://bdcd4786a616a853b8a041f53496d853c68d99d54ff305615cd91c03cd56895e0a7f6e9f35dbf89131044e2114a9a782b792b5661e3aff07faf125a98606a071@43.200.206.40:30303,enode
://209aaf7ed549cf4a5700fd833da25413f80a1248bd3aa7fe2a87203e3f7b236dd729579e5c8df61c97bf508281bae4969d6de76a7393bcbd04a0af70270333b3@54.216.248.9:30303
````
Concensus Layer: heimdall:1.0.1

Concensus Layer Command (with flags/config):
```
start
 --home","/heimdall-home
```
Chain/Network:  mumbai

#### Actual behaviour
````
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x16e4986]

goroutine 307 [running]:
github.com/ledgerwatch/erigon/eth/borfinality.borVerify({0x2f36d50?, 0xc00418c660?}, 0xc0003700e0, 0x26dbc03, 0x26dbf02, {0xc001cf03c2, 0x40}, 0x1)
	github.com/ledgerwatch/erigon/eth/borfinality/bor_checkpoint_verifier.go:78 +0x1a6
github.com/ledgerwatch/erigon/eth/borfinality.fetchWhitelistCheckpoint({0x2f36d50, 0xc00418c660}, {0x2f48e88?, 0xc001921da0?}, 0xc000765e48, 0x533fca?)
	github.com/ledgerwatch/erigon/eth/borfinality/whitelist_helpers.go:45 +0x4fc
github.com/ledgerwatch/erigon/eth/borfinality.handleWhitelistCheckpoint({0x2f36d50?, 0xc00418c660?}, {0x2f48e88?, 0xc001921da0?}, 0x0?)
	github.com/ledgerwatch/erigon/eth/borfinality/whitelist.go:152 +0x75
github.com/ledgerwatch/erigon/eth/borfinality.retryHeimdallHandler(0x2c1e3e0, 0xc0003700e0, 0x4000?, 0x400011040000820?, {0x2805656, 0x14})
	github.com/ledgerwatch/erigon/eth/borfinality/whitelist.go:118 +0xdb
github.com/ledgerwatch/erigon/eth/borfinality.RetryHeimdallHandler(...)
	github.com/ledgerwatch/erigon/eth/borfinality/whitelist.go:100
github.com/ledgerwatch/erigon/eth/borfinality.startCheckpointWhitelistService(0x108?)
	github.com/ledgerwatch/erigon/eth/borfinality/whitelist.go:65 +0x45
created by github.com/ledgerwatch/erigon/eth/borfinality.Whitelist
	github.com/ledgerwatch/erigon/eth/borfinality/whitelist.go:46 +0x225
````
#### Steps to reproduce the behaviour
Restart erigon in any time since stage 5

**Comment by Johnaverse** [comment]
I got the same issue with `--bor.milestone=true` flag

**Comment by mikolajsobolewski** [comment]
seems fixed in 2.50.2
